src/formatDate.js

```diff
@@ -7,8 +7,50 @@
  *
  * @returns {string}
  */
-function formatDate(date, fromFormat, toFormat) {
-  // write code here
+function formatDate(date, sourceFormat, targetFormat) {
+  // Делаем деструктуризацию кода
+  const [, , , sourceSeparator] = sourceFormat;
+  const [, , , targetSeparator] = targetFormat;
+
+  // Разделение даты по сепаратору
+  const dateParts = date.split(sourceSeparator);
+  const [sourceYear, sourceMonth, sourceDay] = dateParts;
+
+  const formattedDateParts = [];
+
+  // Переформатирование даты в зависимости от исходного формата
+  if (
+    sourceFormat[0] === 'YYYY' &&
+    targetFormat[0] === 'DD' &&
+    targetFormat[1] === 'MM' &&
+    targetFormat[2] === 'YYYY'
+  ) {
+    formattedDateParts.push(sourceDay, sourceMonth, sourceYear);
+  } else if (targetFormat[2] === 'YY' && sourceFormat[2] === 'YYYY') {
+    formattedDateParts.push(
+      dateParts[0],
+      dateParts[1],
+      dateParts[2].slice(2, 4),
```

Comment:
This condition only checks for the year conversion from 'YYYY' to 'YY'. However, it does not consider the position of the year in the target format. You need to ensure that the year is placed in the correct position according to the target format.